src/analyses/persistence/settings.clj

```diff
@@ -0,0 +1,83 @@
+(ns analyses.persistence.settings
+  (:use [analyses.persistence.common :only [de log-statement query]])
+  (:require [clojure.java.jdbc :as jdbc]
+            [clojure.tools.logging :as log]
+            [honeysql.core :as sql]
+            [honeysql.helpers :as h]))
+
+(defn list-concurrent-job-limits
+  "Lists all defined concurrent job limits."
+  []
+  (as-> (h/select [:launcher :username] :concurrent_jobs) q
+    (h/from q :job_limits)
+    (h/order-by q [:launcher :asc])
+    (log-statement q)
+    (query q)))
+
+(defn- get-concurrent-job-limit*
+  "Gets the concurrent job limit for a user."
+  [tx username]
+  (as-> (h/select [:launcher :username] :concurrent-jobs) q
+    (h/from q :job_limits)
+    (h/where q [:= :launcher username])
+    (sql/format q)
+    (log/spy :debug q)
```

Comment:
Just a newb question - what's the difference between `log/debug` and `log/spy :debug`?

Reply:
There are two primary differences that I can think of offhand. The first difference is that `log/spy` returns the value that's being logged, which makes it really useful for debug level logging. The other difference is that it includes the expression that is being logged in the log message itself so that you don't have to do things like this:

``` clojure
(log/debug "foo:" foo)
```

Here's an example use of `log/spy`:

```
analyses.core=> (log/spy :debug foo)
2020-04-07 10:52:05,205 DEBUG [nRepl-session-6e324a69-6fb0-449f-9fe3-bf04b64758dc] analyses.core: foo
=> {:bar "baz", :quux "bang"}
```

The output in the example above is the actual message that's being logged. I omitted the return value that's printed by the REPL. I tend to use `log/spy` (usually using a logging level of `:warn`) a lot when I'm debugging. It's a really handy way to figure out what's going in in the program.